server/methods/setAvatarFromService.coffee

```diff
@@ -11,6 +11,28 @@ Meteor.methods
 			RocketChat.models.Users.setAvatarOrigin user._id, service
 			return
 
+		if service is 'url'
+			try
+				result = HTTP.get dataURI, npmRequestOptions: {encoding: 'binary'}
+
+				if result.statusCode isnt 200
+					console.log "Not a valid response, #{result.statusCode}, from the avatar url: #{dataURI}"
+					throw new Meteor.Error('invalid-avatar-url', '[methods] setAvatarFromService -> url service -> error on getting the avatar from url')
+
```

Comment:
Sure, but IMHO you should check for the first part of the data type `/image\/.+/` at least